exonum-java-binding/launcher-plugins/tests/test_instance_plugin.py

```diff
@@ -0,0 +1,46 @@
+import os
+import unittest
+
+from exonum_launcher.configuration import Configuration
+from exonum_launcher.instances import InstanceSpecLoadError
+
+from exonum_java_instance_plugin import JavaInstanceSpecLoader
+
+_DIR_PATH = os.path.dirname(os.path.realpath(__file__))
+
+
+class TestInstancePlugin(unittest.TestCase):
+    @staticmethod
+    def load_config(file_name: str) -> Configuration:
+        """Loads Configuration from the sample .yml file"""
+        config_path = os.path.join(_DIR_PATH, "test_data", file_name)
+        return Configuration.from_yaml(config_path)
+
+    def test_plugin_encodes_config_correctly(self) -> None:
+        config = self.load_config("instance_plugin.yml")
+        self.assertEqual(len(config.instances), 1)
+        instance_loader = JavaInstanceSpecLoader()
+
+        for instance in config.instances:
+            spec = instance_loader.load_spec(None, instance)
```

Comment:
What is `spec`? There are a handful of various "specs" in the deploy & start process. Is it an initial configuration?

The core we use uses `ArtifactSpec.payload` field name, which could be improved, I think (initial_config?): exonum-java-binding/common/src/main/proto/src/runtime.proto:45

Reply:
🤔 Isn't the `ArtifactSpec.payload` contain binary representation of the _artifact_ (in our case, a string containing filename of the artifact)?

InstanceLoader.load_spec returns a serialized initial service configuration.